src/object/sprite_particle.hpp

```diff
@@ -39,6 +39,8 @@ class SpriteParticle final : public GameObject
                  int drawing_layer = LAYER_OBJECTS-1, bool notimeout = false, Color color = Color::WHITE);
   ~SpriteParticle() override;
 
+  SpritePtr& get_sprite() { return sprite; }
```

Comment:
Why is this a SpritePtr reference?

Reply:
Why not?

Reply:
You can make this const if you return `SpritePtr` (which is `std::unique_ptr<Sprite>`) instead of `SpritePtr&`. It doesn't change anything.

Reply:
Maybe `Sprite& get_sprite() { return *sprite; }` would be better?

Reply:
> You can make this const if you return `SpritePtr` (which is `std::unique_ptr<Sprite>`) instead of `SpritePtr&`. It doesn't change anything.

Doing it this way creates many errors.

Reply:
"creates many errors"...WHAT errors?

Reply:
It's a `unique_ptr`, so copy issues I'd assume.

Reply:
Many copies of this:
"error C2280: 'std::unique_ptr<Sprite,std::default_delete<Sprite>>::unique_ptr(const std::unique_ptr<Sprite,std::default_delete<Sprite>> &)': attempting to reference a deleted function"

Reply:
Or maybe return a `SpritePtr`instance by using std::unique_ptr::get()